src/core/grpc_service.proto

```diff
@@ -370,47 +370,69 @@ message InferSharedMemoryRegion
 //@@
 message SharedMemoryControlRequest
 {
-  //@@  .. cpp:enum:: Type
+  //@@  .. cpp:var:: message RegisterInfo
   //@@
-  //@@     Types of control operations for shared memory
+  //@@     The information needed to register the shared memory region.
   //@@
-  enum Type {
-    //@@    .. cpp:enumerator:: Type::REGISTER = 0
+  message RegisterInfo
+  {
+    //@@  .. cpp:var:: InferSharedMemoryRegion shared_memory_region
     //@@
-    //@@       To register the specified shared memory region.
+    //@@     The shared memory region to register.
     //@@
-    REGISTER = 0;
+    InferSharedMemoryRegion shared_memory_region = 5;
```

Comment:
InferSharedMemoryRegion is no longer needed as a separate mesage. The contents of that message should become the contents of Register.

Reply:
Deleted InferSharedMemoryRegion. Reuses SharedMemoryRegion from server_status.proto

Reply:
This comment seems wrong. I still see InferSharedMemoryRegion. It should be deleted and contents moved to Register. SharedMemoryRegion is the status and in the future we may add more things to the status so don't reuse that here.